packages/v2/contracts/ANTv2.sol

```diff
@@ -0,0 +1,171 @@
+pragma solidity 0.5.17;
+
+import './interfaces/IERC20.sol';
+import './libraries/SafeMath.sol';
+
+
+// Lightweight token modelled after UNI-LP: https://github.com/Uniswap/uniswap-v2-core/blob/v1.0.1/contracts/UniswapV2ERC20.sol
+// Adds:
+//   - An exposed `mint()` with minting role
+//   - An exposed `burn()`
+//   - ERC-3009 (`transferWithAuthorization()`)
+contract ANTv2 is IERC20 {
+    using SafeMath for uint256;
+
+    string public constant name = "Aragon Network Token";
+    string public constant symbol = "ANT";
+    uint8 public constant decimals = 18;
+
+    address public minter;
+    uint256 public totalSupply;
+    mapping (address => uint256) public balanceOf;
+    mapping (address => mapping(address => uint256)) public allowance;
+
+    bytes32 public DOMAIN_SEPARATOR;
+
+    // bytes32 public constant PERMIT_TYPEHASH = keccak256("Permit(address owner,address spender,uint256 value,uint256 nonce,uint256 deadline)");
+    bytes32 public constant PERMIT_TYPEHASH = 0x6e71edae12b1b97f4d1f60370fef10105fa2faae0126114a169c64845d6126c9;
+    mapping (address => uint256) public nonces;
+
+    // bytes32 public constant TRANSFER_WITH_AUTHORIZATION_TYPEHASH =
+    //     keccak256("TransferWithAuthorization(address from,address to,uint256 value,uint256 validAfter,uint256 validBefore,bytes32 nonce)");
+    bytes32 public constant TRANSFER_WITH_AUTHORIZATION_TYPEHASH = 0x7c7c6cdb67a18743f49ec6fa9b35f50d52ed05cbed4cc592e13b44501c1a2267;
+    mapping (address => mapping (bytes32 => bool)) public authorizationState;
+
+    event Approval(address indexed owner, address indexed spender, uint256 value);
+    event Transfer(address indexed from, address indexed to, uint256 value);
+    event AuthorizationUsed(address indexed authorizer, bytes32 indexed nonce);
+    event ChangeMinter(address indexed minter);
+
+    modifier onlyMinter {
+        require(msg.sender == minter, "ANTV2:NOT_MINTER");
+        _;
+    }
+
+    constructor(uint256 chainId, address initialMinter) public {
+        DOMAIN_SEPARATOR = keccak256(
+            abi.encode(
+                keccak256('EIP712Domain(string name,string version,uint256 chainId,address verifyingContract)'),
+                keccak256(bytes(name)),
+                keccak256(bytes('1')),
+                chainId,
+                address(this)
+            )
+        );
+
+        _changeMinter(initialMinter);
+    }
+
+    function _validateSignedData(address signer, bytes32 encodeData, uint8 v, bytes32 r, bytes32 s) internal view {
```

Comment:
Any reason why placing the internal methods at the beginning?

Reply:
Haha yes, I know it's a bit awkward, especially since `ANTv2` and `ANTv2Migrator` use different code styles. Planning on expanding on this in the README, but the goal was to minimize the changes to the token since it was taken from UNI-LP :).